jenkins/ompi/ompi_test.sh

```diff
@@ -110,265 +103,226 @@ mlnx_cov="SKIP"
+    has_ucx=$("${OMPI_HOME}/bin/ompi_info" --param pml all --level 9 | grep --count ucx || true)
+    has_btl_openib=$("${OMPI_HOME}/bin/ompi_info" --param btl all | grep --count openib || true)
 
-        if [ -f "$exe_path" ]; then
+    for hca_dev in $(ibstat -l)
+    do
+        if [ -f "${exe_path}" ]
+        then
             local hca="${hca_dev}:${hca_port}"
-            mca="$common_mca --mca btl_openib_if_include $hca -x UCX_NET_DEVICES=$hca --mca btl_openib_allow_ib true"
+            mca="${common_mca} --mca btl_openib_if_include $hca -x UCX_NET_DEVICES=$hca --mca btl_openib_allow_ib true"
 
-            echo "Running $exe_path ${exe_args}"
+            echo "Running ${exe_path} ${exe_args}"
 
-            if [ $has_btl_openib -gt 0 ] && [ "$btl_openib" == "yes" ] ; then
-                $timeout_exe $mpirun --np $np $mca --mca pml ob1 --mca btl self,openib ${AFFINITY} ${exe_path} ${exe_args}
-                if [ "$jenkins_test_xrc" = "yes" ] ; then
-                    $timeout_exe $mpirun --np $np $mca --mca pml ob1 --mca btl self,openib --mca btl_openib_receive_queues X,4096,1024:X,12288,512:X,65536,512 \
-                        ${AFFINITY} ${exe_path} ${exe_args}
+            if [ "${has_btl_openib}" -gt 0 ] && [ "${btl_openib}" = "yes" ]
+            then
+                # shellcheck disable=SC2086
+                ${timeout_exe} "$mpirun" --np "$np" $mca --mca pml ob1 --mca btl self,openib ${AFFINITY} "${exe_path}" "${exe_args}"
+
+                if [ "${ci_test_xrc}" = "yes" ]
+                then
+                    # shellcheck disable=SC2086
+                    ${timeout_exe} "$mpirun" --np "$np" $mca --mca pml ob1 --mca btl self,openib --mca btl_openib_receive_queues X,4096,1024:X,12288,512:X,65536,512 \
+                        ${AFFINITY} "${exe_path}" "${exe_args}"
                 fi
             fi
 
-#if [ "$hca_dev" = "mlx4_0" ]; then
-#                rdma_opt="-mca btl_openib_receive_queues P,65536,256,192,128:S,128,256,192,128:S,2048,1024,1008,64:S,12288,1024,1008,64:S,65536,1024,1008,64"
-#                $timeout_exe $mpirun -np $np $common_mca $rdma_opt -mca btl_openib_cpc_include rdmacm -mca pml ob1 -mca btl self,openib -mca btl_openib_if_include ${hca_dev}:2 \
-#                ${AFFINITY} ${exe_path} ${exe_args}
-#            fi
-
-            if [ "$jenkins_test_ucx" = "yes" -a $has_ucx -gt 0 -a "$hca_dev" != "mlx4_0" ]; then
-                $timeout_exe $mpirun --np $np $mca --mca btl self --mca pml ucx ${AFFINITY} ${exe_path} ${exe_args}
-            fi
-            if [ $has_yalla -gt 0 ]; then
-                $timeout_exe $mpirun --np $np $mca --mca btl self --mca pml yalla ${AFFINITY} ${exe_path} ${exe_args}
-            fi
-            if [ -n "$mpi_custom_args" ]; then
-                $timeout_exe $mpirun --np $np $mca --mca btl self $mpi_custome_args ${AFFINITY} ${exe_path} ${exe_args}
+            if [ "${ci_test_ucx}" = "yes" ] && [ "${has_ucx}" -gt 0 ] && [ "${hca_dev}" != "mlx4_0" ]
```

Comment:
remove `if` only, keep the body by default ON

Reply:
Fixed.